polkadot/runtime/parachains/src/paras/mod.rs

```diff
@@ -653,6 +663,20 @@ pub mod pallet {
 		///
 		/// TODO: Remove once coretime is the standard across all chains.
 		type AssignCoretime: AssignCoretime;
+
+		/// The fungible instance used by the runtime.
+		type Fungible: Mutate<Self::AccountId, Balance: From<BlockNumberFor<Self>>>;
+
+		/// Multiplier to determine the cost of removing upgrade cooldown.
+		///
+		/// After a parachain upgrades their runtime, an upgrade cooldown is applied
+		/// ([`configuration::HostConfiguration::validation_upgrade_cooldown`]). This cooldown
+		/// exists to prevent spamming the relay chain with runtime upgrades. But as life is going
+		/// on, mistakes can happen and a consequent may be required. The cooldown period can be
+		/// removed by using [`Pallet::remove_upgrade_cooldown`]. This dispatchable will use this
+		/// multiplier to determine the cost for removing the upgrade cooldown. Time left for the
+		/// cooldown multiplied with this multiplier determines the cost.
+		type CooldownRemovalMultiplier: Get<BalanceOf<Self>>;
```

Comment:
as you mentioned, the upgrade cooldown is dynamically configured, where's this parameter is static, but implicitly depends on the upgrade cooldown for it's configuration. that means whoever will propose a change to the former, needs to be aware of that implicit dependency.

I suggest an implementation that dynamically reads this cooldown value from the config instead. So this param would not need to be `/ DAYS`.

Reply:
I don't get why you need to depend on it implicitly? If the costs are 1000 per day and the cooldown is 2 days, it would cost 2000 at the beginning.